### DISCUSSION TITLE
Invoice template for Excel - Looks like I need .xlsx.twig. Where do I find it?

### DISCUSSION THREAD

**Discussion by robertrackl** [kimai/kimai#4519] [Questions & Answers] [answered]
I created an Excel invoice template for my purposes; called it pnw_export.xlsx. Stored it in ~/var/invoices in my self-hosted installation. But it does not show up in the dropdown list for selecting invoice templates in the create invoice template dialog, even after resetting the cache. Do I also need a .xlsx.twig file because no such file exists in ~/templates/invoice/renderer ? If so, how do I get it, please?

**Comment by kevinpapst** [comment]
I created `1-foobar.xlsx` in `var/invoices/`:
<img width="312" alt="Bildschirmfoto 2024-01-03 um 14 43 51" src="https://github.com/kimai/kimai/assets/533162/d5acf392-a759-43ea-b926-a0987c76e6b5">

And it shows up under the header `Excel`:
![image](https://github.com/kimai/kimai/assets/533162/03bcbcc9-588d-4b0d-b009-95ec4c4511d9)

That works out of the box, no plugin required

**Reply by robertrackl** [reply]
Hmm ... works for you but not for me. Here is a picture of file manager in my cPanel:
![CPanelInvoiceFiles](https://github.com/kimai/kimai/assets/7914848/130b8d4c-dd24-4cf1-bc38-57db41c7d2a7)
Here is a picture of what I see when I try to create an invoice:
![image](https://github.com/kimai/kimai/assets/7914848/bd73bf3f-6fc0-4820-8193-1e764930cd23)
i.e., there is no .csv or .xlxs file listed even though they exist in file manager in var/invoices. I wonder what I am missing ...?
Also, I reloaded the cache again:
![image](https://github.com/kimai/kimai/assets/7914848/8acc173d-ff91-41e5-842c-b05e21709fb0)
which did not help the situation. Probably makes no difference, but my installation is at GreenGeeks.com.

**Reply by kevinpapst** [reply]
permission problem? maybe kimai can't see the files. can you 777 the entire var/invoices directory and the file itself and retry

**Reply by robertrackl** [reply]
Did as you suggested:
![image](https://github.com/kimai/kimai/assets/7914848/d91442fc-be9c-4f5d-ae11-7199eb0f94be)
Directories var and invoices as well. Reloaded cache. Still does not work. I am perplexed.

**Reply by kevinpapst** [reply, marked as answer]
You have read that it is NOT supported to have two files with the same name?
- remove both files
- try the upload form in Kimai directly (Invoice Templates > Upload)
<img width="364" alt="Bildschirmfoto 2024-01-03 um 21 52 16" src="https://github.com/kimai/kimai/assets/533162/440e6253-0a4a-4f34-9725-be74f17413fa">

**Reply by robertrackl** [reply]
Many thanks for your corrections. It works now.
